signer/tests/integration/transaction_coordinator.rs

```diff
@@ -1791,6 +1784,672 @@ async fn sign_bitcoin_transaction() {
+
+    assert_eq!(deposit_tx.compute_txid(), deposit_request.outpoint.txid);
+
+    let body = deposit_request.as_emily_request();
+    let _ = deposit_api::create_deposit(emily_client.config(), body)
+        .await
+        .unwrap();
+
+    // =========================================================================
+    // Step 7 - Confirm the deposit and wait for the signers to do their
+    //          job.
+    // -------------------------------------------------------------------------
+    // - Confirm the deposit request. This will trigger the block observer
+    //   to reach out to Emily about deposits. Emily will have one so the
+    //   signers should do basic validations and store the deposit request.
+    // - Each TxSigner process should vote on the deposit request and
+    //   submit the votes to each other.
+    // - The coordinator should submit a sweep transaction. We check the
+    //   mempool for its existence.
+    // =========================================================================
+    faucet.generate_blocks(1);
+
+    wait_for_signers(&signers).await;
+
+    let (ctx, _, _, _) = signers.first().unwrap();
+    let mut txids = ctx.bitcoin_client.inner_client().get_raw_mempool().unwrap();
+    assert_eq!(txids.len(), 1);
+
+    let block_hash = faucet.generate_blocks(1).pop().unwrap();
+
+    wait_for_signers(&signers).await;
+
+    // Now lets check the bitcoin transaction, first we get it.
+    let txid = txids.pop().unwrap();
+    let tx_info = ctx
+        .bitcoin_client
+        .get_tx_info(&txid, &block_hash)
+        .unwrap()
+        .unwrap();
+    // We check that the scriptPubKey of the first input is the signers'
+    let actual_script_pub_key = tx_info.vin[0].prevout.script_pub_key.script.as_bytes();
+
+    assert_eq!(actual_script_pub_key, script_pub_key1.as_bytes());
+    assert_eq!(&tx_info.tx.output[0].script_pubkey, &script_pub_key1);
+
+    // Now we check that out database has the sweep transaction
+    let tx = sqlx::query_scalar::<_, BitcoinTx>(
+        r#"
+        SELECT tx
+        FROM sbtc_signer.transactions
+        WHERE txid = $1
+        "#,
+    )
+    .bind(txid.to_byte_array())
+    .fetch_one(ctx.storage.pool())
+    .await
+    .unwrap();
+
+    let script = tx.output[0].script_pubkey.clone().into();
+    for (_, db, _, _) in signers.iter() {
+        assert!(db.is_signer_script_pub_key(&script).await.unwrap());
+    }
+
+    // =========================================================================
+    // Step 9 - Run DKG Again
+    // -------------------------------------------------------------------------
+    // - "Hide" the DKG shares from the database so that the signers think
+    //   that they need to run DKG again.
+    // - After DKG has run a second time, "reveal" the old DKG shares. We
+    //   will use them later.
+    // =========================================================================
+    for (_, db, _, _) in signers.iter() {
```

Comment:
Hm, I might have opted to add a new method to `TestContext` that lets you update the config in-flight, then you could use the configuration variables and we'd also be testing that logic. 🤷

Reply:
Yeah, good point. Updated to use the new configuration variables in https://github.com/stacks-network/sbtc/pull/1210/commits/7a3887bf312acd5685baaece2d7f966d76049c2a. We need to do block accounting upfront, but it works well.